odoo/addons/base/res/res_currency.py

```diff
@@ -169,34 +174,38 @@ def is_zero(self, amount):
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency):
+    def _get_conversion_rate(self, from_currency, to_currency, company_id, date):
```

Comment:
same as above: avoid record ids and use records instead (especially when other parameters are records)